dialogue-java-client/src/main/java/com/palantir/dialogue/HttpChannel.java

```diff
@@ -58,7 +58,7 @@ private HttpChannel(HttpClient client, URL baseUrl, Duration requestTimeout) {
                 null == Strings.emptyToNull(baseUrl.getUserInfo()), "baseUrl user info must be empty");
         this.baseUrl = UrlBuilder.withProtocol(baseUrl.getProtocol())
                 .host(baseUrl.getHost())
-                .port(baseUrl.getPort());
+                .port(getPortOrDefault(baseUrl));
```

Comment:
Can we build this into the urlbuilder?

Reply:
sure, at that point we might as well put all of the logic for parsing a URL in URLBuilder

Reply:
That would be fantastic. I think the api is a bit dangerous currently because it’s mutable and we must remember to call newBuilder before mutating the instance for each request.

Perhaps initialize from a url, and render given a request and endpoint? That should simplify all our clients.